crates/tests/src/e2e/ledger_tests.rs

```diff
@@ -2808,3 +2820,407 @@ fn test_genesis_manipulation() -> Result<()> {
+            "--gas-limit",
+            "2200000",
+            "--node",
+            &validator_one_rpc,
+        ]);
+        let mut client =
+            run!(test, Bin::Client, submit_proposal_args, Some(40))?;
+        client.exp_string(TX_APPLIED_SUCCESS)?;
+        client.assert_success();
+
+        // Start the voting epoch
+        let _epoch = epoch_sleep(&test, &validator_one_rpc, 120)?;
+
+        let proposal_id_str = proposal_id.to_string();
+        let submit_proposal_vote = apply_use_device(vec![
+            "vote-proposal",
+            "--proposal-id",
+            &proposal_id_str,
+            "--vote",
+            "yay",
+            "--address",
+            "validator-0",
+            "--node",
+            &validator_one_rpc,
+        ]);
+
+        let mut client =
+            run!(test, Bin::Client, submit_proposal_vote, Some(40))?;
+        client.exp_string(TX_APPLIED_SUCCESS)?;
+        client.assert_success();
+
+        proposal_id += 1;
+
+        Ok(activation_epoch)
+    };
+
+    // Propose phase 2 - staking party 🥳
+    let activation_epoch = submit_proposal(TestWasms::TxProposalPhase2)?;
+
+    // Wait for phase 2 proposal activation
+    let mut current_epoch = get_epoch(&test, &validator_one_rpc)?;
+    while current_epoch < activation_epoch {
+        // There should be no staking rewards before phase 2
+        let staking_rewards =
+            query_staking_rewards(&test, "validator-0", &validator_one_rpc)?;
+        assert_eq!(staking_rewards, token::Amount::zero());
+
+        // There should be no tokens in PGF address
+        let balance = find_balance(
+            &test,
+            Who::Validator(0),
+            NAM,
+            PGF_ADDRESS,
+            None,
+            None,
+        )?;
+        assert_eq!(balance, token::Amount::zero());
+
+        current_epoch = epoch_sleep(&test, &validator_one_rpc, 120)?;
+    }
+
+    // Check staking rewards activation
+    let staking_rewards =
+        query_staking_rewards(&test, "validator-0", &validator_one_rpc)?;
+    assert_ne!(staking_rewards, token::Amount::zero());
+
+    // There should be some balance now in PGF address
+    let balance =
+        find_balance(&test, Who::Validator(0), NAM, PGF_ADDRESS, None, None)?;
+    assert_ne!(balance, token::Amount::zero());
+
+    // Propose phase 3 - shielding party 🥳
+    let activation_epoch = submit_proposal(TestWasms::TxProposalPhase3)?;
+
+    let namada_receiver = find_address(&test, ALBERT)?.to_string();
+    let ibc_denom_on_namada =
+        format!("{port_id_namada}/{channel_id_namada}/{GAIA_COIN}");
+    let gaia_token = ibc_token(&ibc_denom_on_namada).to_string();
+
+    // IBC transfers shouldn't be allowed before phase 3
+    ibc_tests::transfer_from_gaia(
+        &test_gaia,
+        GAIA_USER,
+        &namada_receiver,
+        GAIA_COIN,
+        1,
+        &port_id_gaia,
+        &channel_id_gaia,
+        None,
+        None,
+    )?;
```

Comment:
does this  function ensure that the transfer fails?

Reply:
this fn doesn't fail - it gets submitted to the relayer where it hangs before timing out (or if the next phase begins first, it goes through). The balance check below that ensures that the transfer doesn't get completed before the next phase is activated